frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2212,6 +2212,21 @@ doIsCandidateApplicableInitial(Context* context,
     }
   }
 
+  if (ci.isMethodCall() && ci.name() == "init") {
+    // TODO: test when record has defaults for type/param fields
+    auto recv = ci.calledType();
+    auto fn = parsing::idToAst(context, candidateId)->toFunction();
+    ResolutionResultByPostorderID r;
+    auto vis = Resolver::createForInitialSignature(context, fn, r);
+    fn->thisFormal()->traverse(vis);
+    auto res = vis.byPostorder.byAst(fn->thisFormal());
```

Comment:
Are we guaranteed to have a this formal? What if I make an `init` proc that isn't a method?

Reply:
This chunk of code is within an if-statement that check if we're dealing with a method, so it should be fine.